gapic/src/main/com/google/gapid/perfetto/models/VSync.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gapid.perfetto.models;
+
+import static com.google.gapid.perfetto.models.QueryEngine.expectOneRow;
+import static com.google.gapid.util.MoreFutures.transform;
+import static java.lang.String.format;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
+/**
+ * A source of V-Sync data.
+ */
+public interface VSync {
+  public static final VSync EMPTY = new VSync() {
+    @Override
+    public boolean hasData() {
+      return false;
+    }
+
+    @Override
+    public Data getData(Track.DataRequest req, Track.OnUiThread<Data> onUiThread) {
+      return new Data(req, new long[0], false);
+    }
+  };
+
+  public default boolean hasData() {
+    return true;
```

Comment:
Is it good to assume true by default? 
Would it be better to have default being false, and for class FromSurfaceFlingerAppCounter judging the boolean value by saved CounterInfo?

Reply:
The idea here is that the only instance that should return false for `hasData` is the `VSync.EMPTY` instance. If the counter doesn't exist, we should not return a `FromSurfaceFlingerAppCounter` instance. This is in preparation of having a better data source for future versions of Android (from surfaceflinger events in R, for example).

Reply:
Thanks for explaining, then it's good to go now. We can take care of this method in the future if necessary, considering how the new data source operates.